client/my-sites/stats/stats-purchase/tier-upgrade-slider/index.tsx

```diff
@@ -58,10 +58,25 @@ function TierUpgradeSlider( {
 		return <div { ...props }>{ thumbSVG }</div>;
 	} ) as RenderThumbFunction;
 
+	// We need at least two steps for the slider to work.
+	let errorMessage = null;
+	const maxIndex = steps.length - 1;
+	if ( maxIndex < 1 ) {
+		errorMessage = <p>Slider has not been configured properly.</p>;
+	}
+
+	// Bounds check the initial index value.
+	let initialIndex = Math.floor( initialValue );
```

Comment:
I'm having problems understanding the check between lines 61 and 74. 
Is it used to check if the component consumer provided a proper `initialValue` that matches `steps` or `marks`?
This is the responsibility of the developers - component consumers. Users can't input values outside the slider range.
Normally, a check for a value range would be a good practice, but keep in mind that this check will be performed on the client, so every time users use the component this check will be executed and also on every re-render. 

If you want to check if there are at least two slider points, a simple length check would be enough.

Reply:
> Is it used to check if the component consumer provided a proper `initialValue` that matches `steps` or `marks`?

The thinking was to make sure we have at least two steps for the slider and that the initial value is a valid index. As has come up in Slack, we need to allow for the use case of no slider though, so the check no longer makes sense with that consideration in mind.

I keep the other comments in mind for follow-up PRs as we finish up our testing over the next few days.